src/dateUtils.js

```diff
@@ -26,3 +26,7 @@ export function getUserDateFormat() {
 export function getUserDateTimeFormat() {
   return getProcessMakerUserValue('datetime_format') || 'MM/DD/YYYY h:mm A';
 }
+
+export function isValidDate(date, format = getUserDateFormat()) {
+  return moment(date, format).isValid();
```

Comment:
I'm a bit torn about this function. One one level it seems to be a middleman but on another it abstracts away the dependency on moment. What are your thoughts?

Reply:
Good point! The way it's called is, it's passed as a function to the validator:
```
Validator.register('dateFormat', isValidDate, `The :attribute is not a valid date with format ${getUserDateFormat()}`);
```
I'm not sure how I can un-wrap this, as I can't simply pass `moment.isValid`, since it needs to be called from a moment object, and I also need a way to pass the date format. So the best I could do is inline the method, and then test it by calling `validator.passes()`? But we're not really concerned about the validator class.

So I'm fine leaving it like this for now.

Reply:
I think this is fine for now. At the least the abstraction over moment is useful.